File: app/src/main/java/de/westnordost/osmagent/data/download/WifiAutoDownloadStrategy.java
package de.westnordost.osmagent.data.download;

import android.content.SharedPreferences;

import javax.inject.Inject;

import de.westnordost.osmagent.data.QuestTypes;
import de.westnordost.osmagent.data.osm.persist.OsmQuestDao;
import de.westnordost.osmagent.data.tiles.DownloadedTilesDao;

public class WifiAutoDownloadStrategy extends AActiveRadiusStrategy
{
	@Inject public WifiAutoDownloadStrategy(OsmQuestDao osmQuestDB,
											DownloadedTilesDao downloadedTilesDao,
											QuestTypes questTypes, SharedPreferences prefs)
	{
		super(osmQuestDB, downloadedTilesDao, questTypes, prefs);
	}

	/** Let's assume that if the user is on wifi, he is either at home, at work, in the hotel, at a
	 *  café,... in any case, somewhere that would act as a "base" from which he can go on an
	 *  excursion. Let's make sure he can, even if there is no or bad internet.
	 *
	 *  Since download size is almost unlimited, we can be very generous here.
	 *  However, Overpass is as limited as always, so the number of quest types we download is
	 *  limited as before */

	@Override public int getQuestTypeDownloadCount()
	{
		return 3;
	}

	@Override protected int getMinQuestsInActiveRadiusPerKm2()
	{
		return 12;
	}

	@Override protected int getActiveRadius()
	{
		return 1200;
	}

	@Override protected int getDownloadRadius()
	{
		return 2400;
	}
}

File: app/src/main/java/de/westnordost/streetcomplete/data/download/strategy/WifiAutoDownloadStrategy.kt
package de.westnordost.streetcomplete.data.download.strategy

import de.westnordost.streetcomplete.data.download.tiles.DownloadedTilesSource
import de.westnordost.streetcomplete.data.osm.mapdata.MapDataController

/** Download strategy if user is on wifi */
class WifiAutoDownloadStrategy(
    mapDataController: MapDataController,
    downloadedTilesSource: DownloadedTilesSource
) : AVariableRadiusStrategy(mapDataController, downloadedTilesSource) {

    /** Let's assume that if the user is on wifi, he is either at home, at work, in the hotel, at a
     * café,... in any case, somewhere that would act as a "base" from which he can go on an
     * excursion. Let's make sure he can, even if there is no or bad internet.
     */

    override val maxDownloadAreaInKm2 = 20.0 // that's a radius of about 2.5 km
    override val desiredScoredMapDataCountInVicinity = 15000
}
